gobblin-hive-registration/src/main/java/gobblin/hive/policy/HiveRegistrationPolicyBase.java

```diff
@@ -141,33 +211,45 @@ protected String sanitizeAndValidateName(String name) {
   }
 
   /**
-   * A base implementation for creating a non bucketed, external {@link HiveTable} given a {@link Path}.
+   * A base implementation for creating non bucketed, external {@link HiveTable}s given a {@link Path}.
+   *
+   * <p>
+   *   This method returns a list of {@link Hivetable}s that contains one table per db name
+   *   (returned by {@link #getDatabaseNames(Path)}) and table name (returned by {@link #getTableNames(Path)}.
+   * </p>
    *
    * @param path a {@link Path} used to create the {@link HiveTable}.
-   * @return a {@link HiveTable} for the given {@link Path}.
+   * @return a list of {@link HiveTable}s for the given {@link Path}.
    * @throws IOException
    */
-  protected HiveTable getTable(Path path) throws IOException {
-    HiveTable table = new HiveTable.Builder().withDbName(getDatabaseName(path)).withTableName(getTableName(path))
-        .withSerdeManaager(HiveSerDeManager.get(this.props)).build();
-
-    table.setLocation(getTableLocation(path));
-    table.setSerDeProps(path);
-    table.setProps(this.props.getTablePartitionProps());
-    table.setStorageProps(this.props.getStorageProps());
-    table.setSerDeProps(this.props.getSerdeProps());
-    table.setNumBuckets(-1);
-    table.setBucketColumns(Lists.<String> newArrayList());
-    table.setTableType(TableType.EXTERNAL_TABLE.toString());
-    return table;
+  protected List<HiveTable> getTables(Path path) throws IOException {
+    List<HiveTable> tables = Lists.newArrayList();
+
+    for (String databaseName : getDatabaseNames(path)) {
```

Comment:
This will do a cross product of all dbs and tables, so registering only `dba.tablea`, `dbb.tableb` is impossible. A suggestion is for users to instead provide the additional registrations as complete names, where db and table are separated by a dot.